### ISSUE TITLE
[WASM| call_static_method does not support passing arrays

### ISSUE THREAD

**Issue by pranavkm** [mono/mono#18245]
1. Start with the sample produced from the Mono WASM build.
2. Update sample.cs produced from WASM artifacts to have a method that accepts a string array (. For example:

```C#
public class StringTest {
  public static void PrintArrayLength (int[] x) {
    Console.WriteLine(x.Length);
  }
}
```

Invoke the method from JS as follows:

```
BINDING.call_static_method("[sample] Math:PrintArrayLength", [[1, 2] ]); // This produces 19 as the result
```

Inspecting the type of "x" in `PrintArrayLength` produces odd results depending on the JavaScript values that were passed in. For instance using `[[1,2]]` results in the type being `WebAssembly.Core.Array`. If I instead pass in `[1, 2]` the type of `x` is `System.Int32` which seems really strange.

Instead of adding support for arbitrary data types, one option would be to allow passing a managed object wrappers to `call_static_method`

**Comment by MarcosCobena** [comment]
Hi @pranavkm,

I've just stumbled upon this issue with my workmate @emepetres and can share some light on the behavior you see:
- if you pass `[[1, 2]]` you'll receive a `WebAssembly.Core.Array`, which you simply can use as an `object[]`: `(int)myArray[1]` will give you "2", for instance; and
- if you pass `[1, 2]` Mono actually takes the first param, as your signature just receives one param, and this' the reason why you get `System.Int32` :-)

Summing up: just trust on `WebAssembly.Core.Array` for now, and you'll be OK to go:
```
public static void PrintArrayLength (WebAssembly.Core.Array x) {
  Console.WriteLine(x.Length);
}
```

My 2 cents,

—Marcos

**Comment by kjpou1** [comment]
The above is correct that what is passed is a `WebAssembly.Core.Array` if you want to pass an int array then you would need to pass a typed array `(Int32Array or Uint32Array)`.

We do make assumptions of the javascript type passed to the function.  

Will close this issue and thanks to @MarcosCobena for the added information.
